app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/BinaryFilter.java

```diff
@@ -0,0 +1,48 @@
+package com.sdpteam.connectout.event.nearbyEvents.filter;
+
+import com.sdpteam.connectout.event.Event;
+import com.sdpteam.connectout.profile.Profile;
+
+import java.util.List;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+
+/**
+ * Filter that holds both filtering information, on list of participants and events themselves.
+ */
+public class BinaryFilter {
+    private final EventFilter eventFilter;
+    private final ParticipantsFilter participantsFilter;
+
+    public final static BinaryFilter NONE = new BinaryFilter();
+
+    public BinaryFilter(){
+        eventFilter = EventFilter.NONE;
+        participantsFilter = ParticipantsFilter.NONE;
+    }
+    public BinaryFilter(EventFilter eventFilter){
+        this.eventFilter = eventFilter;
+        participantsFilter = ParticipantsFilter.NONE;
+    }
+    public BinaryFilter(EventFilter eventFilter, ParticipantsFilter participantsFilter){
+        this.eventFilter = eventFilter;
+        this.participantsFilter = participantsFilter;
+    }
+
+    public boolean testEvent(Event event){
+        return eventFilter.test(event);
+    }
+    public boolean testParticipants(List<Profile> profiles){
+        return participantsFilter.test(profiles);
+    }
+    public BinaryFilter and(EventFilter eventFilter, ParticipantsFilter participantsFilter) {
```

Comment:
(and wanted to point out that these and or and negate are not used at all, which really shows that this class is not very useful)